zim/plugins/sequencediagrameditor.py

```diff
@@ -41,28 +49,41 @@ class BackwardSequenceDiagramImageObjectType(BackwardImageGeneratorObjectType):
 
 
 class SequenceDiagramGenerator(ImageGeneratorClass):
+	@property
+	def _prefer_svg(self):
+		return self.plugin.preferences['prefer_svg']
+
+	@property
+	def imagefile_extension(self):
+		return '.svg' if self._prefer_svg else '.png'
 
-	imagefile_extension = '.png'
+	@property
+	def diagcmd(self):
+		return diagcmd_svg if self._prefer_svg else diagcmd
 
 	def __init__(self, plugin, notebook, page):
 		ImageGeneratorClass.__init__(self, plugin, notebook, page)
 		self.diagfile = TmpFile('seqdiagram.diag')
 		self.diagfile.touch()
-		self.pngfile = LocalFile(self.diagfile.path[:-5] + '.png') # len('.diag') == 5
 
 	def generate_image(self, text):
 		# Write to tmp file
 		self.diagfile.write(text)
+		self.imgfile = LocalFile(self.diagfile.path[:-5] + self.imagefile_extension) # len('.diag') == 5
```

Comment:
Moved imgfile generation outside of init, otherwise it's buggy when toggling formats in the user prefs.